satellite.py

```diff
@@ -96,3 +149,43 @@
                   if chosen]
 
 print(constellations)
+
+# don't consider constellations with average score less than score_threshold
+score_threshold = .4
+
+bqm = dimod.BinaryQuadraticModel.empty(dimod.BINARY)
+
+# first we want to favor combinations with a high score
+for constellation in itertools.combinations(range(num_satellites), constellation_size):
+    # the score is the average coverage for a constellation
+    score = sum(coverage[v] for v in constellation) / constellation_size
+
+    # to make it smaller, throw out the combinations with a score below
+    # a set threshold
+    if score < score_threshold:
+        continue
+
+    # we subtract the score because we want to minimize the energy
+    bqm.add_variable(frozenset(constellation), -score)
+
+# next we want to penalize pairs that share a satellite. We choose 2 because
+# because we don't want it to be advantageous to pick both in the case that
+# they both have 100% coverage
+for c0, c1 in itertools.combinations(bqm.variables, 2):
+    if c0.isdisjoint(c1):
+        continue
+    bqm.add_interaction(c0, c1, 2)
+
+# finally we wish to choose num_constellations variables. We pick strength of 1
+# because we don't want it to be advantageous to violate the constraint by
+# picking more variables
+bqm.update(dimod.generators.combinations(bqm.variables, num_constellations, strength=1))
```

Comment:
This duplicated code should be moved to a function.

Reply:
It turns out that code wasn't doing anything. It's gone. Thanks for noticing it.